src/Tizen.Uix.Tts/Tizen.Uix.Tts/TtsClient.cs

```diff
@@ -243,84 +241,124 @@ public TtsClient()
         /// Event to be invoked when TTS state changes.
         /// </summary>
         /// <since_tizen> 3 </since_tizen>
+        /// <exception cref="InvalidOperationException">This exception can be due to an invalid state.</exception>
+        /// <exception cref="NotSupportedException">This exception can be due to TTS not supported.</exception>
+        /// <exception cref="ArgumentException">This exception can be due to improper value provided while setting the value.</exception>
+        /// <pre>
+        /// The state must be created.
+        /// </pre>
         public event EventHandler<StateChangedEventArgs> StateChanged
         {
             add
             {
                 lock (thisLock)
                 {
-                    _stateDelegate = (IntPtr handle, State previous, State current, IntPtr userData) =>
-                    {
-                        StateChangedEventArgs args = new StateChangedEventArgs(previous, current);
-                        _stateChanged?.Invoke(this, args);
-                    };
-                    TtsError error = TtsSetStateChangedCB(_handle, _stateDelegate, IntPtr.Zero);
-                    if (error != TtsError.None)
+                    if (State.Created != CurrentState)
                     {
-                        Log.Error(LogTag, "Add StateChanged Failed with error " + error);
+                        Log.Error(LogTag, "Add StateChanged Failed with error " + TtsError.InvalidState);
+                        throw ExceptionFactory.CreateException(TtsError.InvalidState);
                     }
-                    else
+                    if (null == _stateChanged)
                     {
-                        _stateChanged += value;
+                        _stateDelegate = (IntPtr handle, State previous, State current, IntPtr userData) =>
+                        {
+                            StateChangedEventArgs args = new StateChangedEventArgs(previous, current);
+                            _stateChanged?.Invoke(this, args);
+                        };
+                        TtsError error = TtsSetStateChangedCB(_handle, _stateDelegate, IntPtr.Zero);
+                        if (TtsError.None != error)
+                        {
+                            Log.Error(LogTag, "Add StateChanged Failed with error " + error);
+                            throw ExceptionFactory.CreateException(error);
+                        }
                     }
+                    _stateChanged += value;
                 }
-
             }
 
             remove
             {
                 lock (thisLock)
                 {
-                    TtsError error = TtsUnsetStateChangedCB(_handle);
-                    if (error != TtsError.None)
+                    if (State.Created != CurrentState)
```

Comment:
Why restrict the state to Created only?
It could be issue, Usually event handler removed when before release a object and did not check any state. because it is natural to c# developer
It is a important to API usability.

I think, this native limitation can be solved with c# layer implementation.